kaniko/executor_docker_test.go

```diff
@@ -28,7 +28,9 @@ func Test_DockerExecutor_Execute(t *testing.T) {
 		},
 	})
 
-	err := executor.Execute(context.Background(), []string{"kaniko-arg1", "kaniko-arg2"})
+	writer := mock.NewWriter()
+	err := executor.Execute(context.Background(), writer, []string{"kaniko-arg1", "kaniko-arg2"})
+	assert.Equal(t, writer.GetCallCount(), 0)
```

Comment:
I don't think it is relevant to test the call count on the writer. What matters is what was actually written, is it possible to test that instead ? I think it would make the test easier to understand at first sight.

In addition, I don't understand why the call count should be 0 here, the `mock.Executor` is supposed to output `"some output"` isn't it ?